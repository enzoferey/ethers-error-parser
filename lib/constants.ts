export const ERROR_CODES = {
  TRANSACTION_RAN_OUT_OF_GAS: "TRANSACTION_RAN_OUT_OF_GAS",
  TRANSACTION_UNDERPRICED: "TRANSACTION_UNDERPRICED",
  REJECTED_TRANSACTION: "REJECTED_TRANSACTION",
  EXECUTION_REVERTED: "EXECUTION_REVERTED",
  NONCE_TOO_LOW: "NONCE_TOO_LOW",
  UNKNOWN_ERROR: "UNKNOWN_ERROR",
} as const;

export const ETHERS_ERROR_CODES = {
  REJECTED_TRANSACTION: 4001,
  REQUIRE_TRANSACTION: -32603,
  ERROR_WHILE_FORMATTING_OUTPUT_FROM_RPC: -32603,
  TRANSACTION_UNDERPRICED: -32000,
  NONCE_EXPIRED: "NONCE_EXPIRED",
  UNPREDICTABLE_GAS_LIMIT: "UNPREDICTABLE_GAS_LIMIT",
};
