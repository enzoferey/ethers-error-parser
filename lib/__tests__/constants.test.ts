import { describe, it, expect } from "vitest";

import * as constants from "../constants";

describe("constants", () => {
  it("should declare the package constants", () => {
    expect(constants).toMatchSnapshot();
  });
});
