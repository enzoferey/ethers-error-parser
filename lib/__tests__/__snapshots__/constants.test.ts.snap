// Vitest Snapshot v1

exports[`constants > should declare the package constants 1`] = `
{
  "ERROR_CODES": {
    "EXECUTION_REVERTED": "EXECUTION_REVERTED",
    "REJECTED_TRANSACTION": "REJECTED_TRANSACTION",
    "TRANSACTION_RUN_OUT_OF_GAS": "TRANSACTION_RUN_OUT_OF_GAS",
    "TRANSACTION_UNDERPRICED": "TRANSACTION_UNDERPRICED",
    "UNKNOWN_ERROR": "UNKNOWN_ERROR",
  },
  "ETHERS_ERROR_CODES": {
    "ERROR_WHILE_FORMATTING_OUTPUT_FROM_RPC": -32603,
    "REJECTED_TRANSACTION": 4001,
    "REQUIRE_TRANSACTION": -32603,
    "TRANSACTION_UNDERPRICED": -32000,
  },
  Symbol(Symbol.toStringTag): "Module",
}
`;
